src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isPresent() && user.get().getPassword()
+                .equals(HashUtil.hashPassword(password, user.get().getSalt()))) {
+            return user.get();
+        }
+        throw new AuthenticationException("Incorrect email or password");
+    }
+
+    @Override
+    public User register(String email, String password) {
+        if (userService.findByEmail(email).isEmpty()) {
+            return userService.add(new User(email, password));
+        }
+        throw new RuntimeException("Error registration, please used other email");
```

Comment:
it is better to throw custom RegistrationException and it also shouldn't extend RuntimeException